etna/models/prophet.py

```diff
@@ -154,6 +162,33 @@ def get_model(self) -> Prophet:
         """
         return self.model
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        try:
+            model_dict = model_to_dict(self.model)
+            is_fitted = True
+        except ValueError:
+            is_fitted = False
+            model_dict = {}
+        del state["model"]
+        state["_is_fitted"] = is_fitted
+        state["_model_dict"] = model_dict
+        return state
+
+    def __setstate__(self, state):
+        local_state = deepcopy(state)
+        is_fitted = local_state["_is_fitted"]
+        model_dict = local_state["_model_dict"]
+        del local_state["_is_fitted"]
+        del local_state["_model_dict"]
+
+        self.__dict__.update(local_state)
+
+        if is_fitted:
```

Comment:
I don't really think that it is a good idea. Now `_create_model` doesn't know anything about serialization. Just handles creation of model using hyperparameters and processing `additional_seasonalities` (I added this method mostly because of them).